darwin/utils.py

```diff
@@ -80,22 +81,32 @@ def find_files(
     """
     if exclude is None:
         exclude = []
-    if not root.is_dir():
-        if (root.suffix in SUPPORTED_IMAGE_EXTENSIONS + SUPPORTED_VIDEO_EXTENSIONS
-                and root.suffix not in exclude):
-            return [root]
-        else:
-            return []
-    files: List[Path] = []
-    for file in root.iterdir():
-        if file.is_dir():
-            if recursive:
-                files += find_files(file, recursive, exclude)
-        else:
+    if files_list is not None:
+        files: List[Path] = []
+        for file in files_list:
+            file = Path(file)
             if (file.suffix in SUPPORTED_IMAGE_EXTENSIONS + SUPPORTED_VIDEO_EXTENSIONS
                     and file.suffix not in exclude):
                 files += [file]
-    return files
+        return files
+    if root is not None:
+        if not root.is_dir():
+            # print ("TODO: when are we supposed to enter here")
+            if (root.suffix in SUPPORTED_IMAGE_EXTENSIONS + SUPPORTED_VIDEO_EXTENSIONS
+                    and root.suffix not in exclude):
+                return [root]
+            else:
+                return []
+        files: List[Path] = []
+        for file in root.iterdir():
+            if file.is_dir():
+                if recursive:
+                    files += find_files(file, recursive, exclude)
+            else:
+                if (file.suffix in SUPPORTED_IMAGE_EXTENSIONS + SUPPORTED_VIDEO_EXTENSIONS
+                        and file.suffix not in exclude):
+                    files += [file]
```

Comment:
I would probably rewrite the `if`s conditions a bit to enhance readability (at least imho)

```python
if file.is_dir() and recursive:
    files += find_files(file, recursive=recursive, exclude=exclude)
elif not file.is_dir() and file.suffix in SUPPORTED_IMAGE_EXTENSIONS + SUPPORTED_VIDEO_EXTENSIONS and file.suffix not in exclude:
    files += [file]
```

(the formatter would surely make a better job in formatting the code though 😁)

Reply:
I agree this function needs rewriting but I do not know all the cases we enter it for now. I'll test all the functionality for the other PR